File: backend/PharmaceuticalSupplyChainSystem/Models/Shipment.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace PharmaceuticalSupplyChainSystem.Models
{
    public class Shipment
    {
        [BsonId]
        [BsonRepresentation(MongoDB.Bson.BsonType.ObjectId)]
        public ObjectId Id { get; set; }
        public string ShipperID { get; set; }
        public string ShipperName { get; set; }
        public string TrackingNumber { get; set; }
        public string CurrentLocation { get; set; }
        public Order[] Orders { get; set; }
        public DateTime EstimatedDelivery { get; set; }
    }
}

File: frontend/src/shared/models/Shipment.ts
import Order from './Order';

export default class Shipment {
  shipperID: string;
  shipperName: string;
  trackingNumber: string;
  currentLocation: string;
  orders: Order[];
  estimatedDelivery: Date;
}

export class AddShipment {
  shipperID: string;
  shipperName: string;
  trackingNumber: string;
  currentLocation: string;
  orderIds: string[];
  estimatedDelivery: Date;
}
